src/overlay/overlay.scss

```diff
@@ -0,0 +1,376 @@
+  min-width: 192px;
+  flex-direction: column;
+  background-color: var(--color-canvas-default);
+  border-radius: $primer-borderRadius-large;
+  box-shadow: var(--color-overlay-shadow);
+  opacity: 1;
+
+  &.Overlay--height-auto {
+    height: auto;
+  }
+
+  &.Overlay--height-xsmall {
+    height: 192px;
+  }
+
+  &.Overlay--height-small {
+    height: 256px;
+  }
+
+  &.Overlay--height-medium {
+    height: 320px;
+  }
+
+  &.Overlay--height-large {
+    height: 432px;
+  }
+
+  &.Overlay--height-xlarge {
+    height: 600px;
+  }
+
+  &.Overlay--width-auto {
+    width: auto;
+  }
+
+  &.Overlay--width-small {
+    min-width: 256px;
+  }
+
+  &.Overlay--width-medium {
+    min-width: 320px;
+  }
+
+  &.Overlay--width-large {
+    min-width: 480px;
+  }
+
+  &.Overlay--width-xlarge {
+    min-width: 640px;
+  }
+
+  &.Overlay--width-xxlarge {
+    min-width: 960px;
+  }
+
+  &.Overlay--motion-scaleFade {
+    @media screen and (prefers-reduced-motion: no-preference) {
+      animation: 200ms cubic-bezier(0.33, 1, 0.68, 1) 0s 1 normal none running motion--scaleFade;
+    }
+  }
+
+  @keyframes motion--scaleFade {
+    0% {
+      opacity: 0;
+      transform: scale(0.5);
+    }
+
+    100% {
+      opacity: 1;
+      transform: scale(1);
+    }
+  }
+}
+
+.Overlay-header {
+  z-index: 1;
+  display: flex;
+  flex-direction: column;
+
+  &.Overlay-header--divided {
+    padding-bottom: $spacer-2;
+    // stylelint-disable-next-line primer/box-shadow
+    box-shadow: inset 0 #{-$border-width} var(--color-border-default);
+  }
+
+  &.Overlay-header--large {
+    .Overlay-headerContentWrap {
+      .Overlay-titleWrap {
+        gap: $spacer-2;
+
+        .Overlay-title {
+          font-size: $h3-size;
+        }
+
+        .Overlay-description {
+          font-size: $body-font-size;
+        }
+      }
+    }
+  }
+
+  .Overlay-headerContentWrap {
+    display: flex;
+    align-items: flex-start;
+    gap: $spacer-2;
+    padding: $spacer-2 $spacer-2 0 $spacer-2;
+
+    .Overlay-actionWrap {
+      display: flex;
+      flex-direction: row;
+      gap: $spacer-2;
+    }
+
+    .Overlay-titleWrap {
+      display: flex;
+      padding: ($spacer-2 * 0.75) 0 ($spacer-2 * 0.75) $spacer-2;
+      flex-direction: column;
+      flex-grow: 1;
+      gap: $spacer-1;
+
+      .Overlay-title {
+        margin: 0;
+        font-size: $body-font-size;
+        font-weight: $font-weight-bold;
+      }
+
+      .Overlay-description {
+        margin: 0;
+        font-size: $font-size-small;
+        font-weight: $font-weight-normal;
+        color: var(--color-fg-muted);
+      }
+    }
+  }
+}
+
+// generic body content slot
+.Overlay-body {
+  flex-grow: 1;
+  padding: $spacer-3;
+  font-size: $body-font-size;
+
+  &.Overlay-body--paddingCondensed {
+    padding: $spacer-2;
+  }
+
+  &.Overlay-body--paddingNone {
+    padding: 0;
+  }
+}
+
+// generic footer slot
+.Overlay-footer {
+  z-index: 1;
+  display: flex;
+  padding: 0 $spacer-3 $spacer-3 $spacer-3;
```

Comment:
I'm noticing an empty non scrollable area under the body, even when the overlay has no footer:

![Screen Shot 2022-04-26 at 15 12 31](https://user-images.githubusercontent.com/293280/165401278-594adfc2-c23b-4f18-ac8e-180e50d71457.png)

Reply:
I can't figure this one out. It seems like the browser will add that little margin when a scrollbar is present, even if the scrollbar is only vertical. Any ideas? 🤔

Reply:
I think an empty `Overlay-footer` is causing that. Should the rendering template hide the footer element in case it's empty?

Reply:
Yes.. _facepalm_ 

I did handle this in the PVC implementation 😄